### PR TITLE
Flashinfer: TopK+TopP sampling from probs

### PR DESCRIPTION
**Pull request by Kh4L**
Recent Flashinfer versions of `top_k_top_p_sampling_from_logits` are much slower on different systems:

 GPU | from logits (tokens/s) | from probs (tokens/s) | Improvement (%)
-- | -- | -- | --
SPARK | 19.78 | 28.50 | 44.1%
GB200 | 144.84 | 216.60 | 49.5%
H100 | 97.19 | 130.65 | 34.4%

### CODE REVIEW

File: vllm/v1/sample/ops/topk_topp_sampler.py [hunk 1]
```diff
@@ -282,8 +282,9 @@ def flashinfer_sample(
         )
     else:
         # Both top-k and top-p.
-        next_token_ids = flashinfer.sampling.top_k_top_p_sampling_from_logits(
-            logits, k, p, deterministic=True
+        probs = logits.softmax(dim=-1, dtype=torch.float32)
+        next_token_ids = flashinfer.sampling.top_k_top_p_sampling_from_probs(
+            probs, k, p, deterministic=True
```
Comment (lgeiger) [inline vllm/v1/sample/ops/topk_topp_sampler.py:287]: This is interesting, does this mean `top_k_mask_logits` is slow for you?
https://github.com/flashinfer-ai/flashinfer/blob/6d19a75a6963b0996b7b2cfe5f9e385c5cc41a66/flashinfer/sampling.py#L1023-L1033

Might be worth also opening an issue on flashinfer since looking at their code, I don't really see a reason for why `top_k_top_p_sampling_from_probs` should be faster.

For reference, in #18608 I saw the opposite behaviour

**Comment by heheda12345** [comment]
CC @mgoin as you were reviwing https://github.com/vllm-project/vllm/pull/18608

**Comment by Kh4L** [comment]
@mgoin I do not have a strong opinion on that and I think it can be discussed in a separate "Issue", but I think it makes sense to switch to this flashinfer stable kernel in the meantime

File: vllm/v1/sample/ops/topk_topp_sampler.py (hunk 1 above)
Comment (Kh4L) [reply vllm/v1/sample/ops/topk_topp_sampler.py:287]: >This is interesting, does this mean top_k_mask_logits is slow for you?

It is with the latest versions of Flashinfer

I believe the the perf changed with Flashinfer updates
